pylint/checkers/variables.py

```diff
@@ -1174,6 +1209,45 @@ def leave_setcomp(self, _: nodes.SetComp) -> None:
         # do not check for not used locals here
         self._to_consume.pop()
 
+    @utils.only_required_for_messages(
+        "missing-timeout",
+    )
+    def visit_call(self, node: nodes.Call) -> None:
+        """Check if the call needs timeout parameter based on package.func_name
+        configured in config.timeout_methods.
+
+        Package uses the self._imports dict variable
+        in order to know the original package name imported
+        """
+        lib_name = ""
+        if isinstance(node.func, nodes.Attribute) and isinstance(
```

Comment:
I'll need to check this myself locally. But the way you determine the library name seems like it could just work with an inference of the `nodes.Name` and checking the `qname` of the import.

Reply:
Yes, it is valid for only built-in libraries or 3rd-party library already installed

But the CIs rarely install all dependencies to run the static checkers like pylint

Another matter is that using the infered qname the name of the method change a little bit and there is not an easy way to find these kind of names without using pylint/astroid

e.g. the infered qname for `from requests import delete` the result is `requests.api.delete`

Even if you are importing with `requests.delete` the name has an `*.api.*`

So, it could be hard to know that is the real name for the user trying to set values in the pylint.cfg for `timeout-methods` parameter

IMHO it is easier to know the name using the same way than it was imported

and even if it is not installed

Reply:
I think we should use the `qname`. It might be a little bit more difficult to set up, but it's arguably more correct. `pylint` depends on dependencies being installed for checkers like `no-member` anyway so not installing dependencies in the environment that `pylint` uses will likely create troubles anyway.

Reply:
I'm not sure about this

I mean, almost all of the current environments running static checkers are not installing extra dependencies
e.g. pre-commit, travis-CI, github actions

So, It could be virtually disabled for almost all of this environment already configured

What about doing work done without installing extra packages?

Reply:
> I mean, almost all of the current environments running static checkers are not installing extra dependencies e.g. pre-commit, travis-CI, github actions

All of those also allow specifying additional dependencies. For example, our own configuration for `mypy` had dependencies:
https://github.com/PyCQA/pylint/blob/0777a0ecb5c730acfa54186c56af508692bec821/.pre-commit-config.yaml#L95-L106
Similarly, in our own run of `pylint` we make sure to install a set of dependencies into the environment:
https://github.com/PyCQA/pylint/blob/0777a0ecb5c730acfa54186c56af508692bec821/.github/workflows/checks.yaml#L58-L59

I understand that creates some additional work for users, but I think that's an inherent limitation to static checkers.

> So, It could be virtually disabled for almost all of these environments already configured

I'd suggest to anybody to install dependencies in environments that use static checkers. Using `caches` this shouldn't even slow down your CI after the initial cache has been created.

> What about doing work done without installing extra packages?

If that was the general policy of `pylint` I'd be fine with that, but we have other checks that do depend on you installing the dependencies. I don't think we should mix that.

Reply:
Most of pylint checks won't work without dependencies installed, it has some disadvantages but also some advantages that are a differentiating factor of pylint compared to other linters. This is a fundamental assumption of the software that can't be changed (unless we want to have the worst of both world).

Reply:
Yes, there are not another way to raise a `no-member` or 

So, it is not a static checker but dynamic one since that it is mutable based on the environment

I mean, it is not possible to define a static checker for `no-member`

Similar case for `import-error`

But it is not the case for `missing-timeout` since that already there is a way to define it as static checker without extra dependencies installed

Similar to `reimported`

You even don't need to `infer` to know if a library was `reimported`

Of course, you could use infer but there is a way to avoid it so use it

So, you will get false positives for reimported if you have:

```python
# file1.py
from os import os as os1

# file2.py
import os
from .file1 import os1  # re-imported
```

But IMHO it is better if you have "plug&play" static checkers

Reply:
I have created a separated PR with inferred:

https://github.com/PyCQA/pylint/pull/6885#issuecomment-1148055447

Notice that the detected cases were less than this PR

I mean,

PR using static checkers is detecting in:
- sentry
- pandas
- pytest

but new PR only in sentry

Reply:
We're going to get issues opened for it if we don't handle this. We're talking about a fundamental assumption that define what pylint is supposed to do since 2003. The expectation from pylint users is that aliases and namespace aliases should be handled correctly, see https://github.com/PyCQA/pylint/pull/6787 for example. We're advertising that [in the very first paragraph of our doc introduction](https://pylint.pycqa.org/en/latest/). It's a differentiating factor of pylint and one of the reason pylint should be used alongside flake8. pylint needs dependencies installed anyway and already is relying on this assumption for other checks so there is no cost to pylint's user of doing it this way and pylint do not function without this assumption. 

There could be a "light mode" pylint with only static analysis but imo it's a lot of work to create a flake8 clone.

Reply:
Fixed in last commit, thank you for your feedback!